guardian/grid
# OIDC upgrades

## What does this change?

Upgrades the dependency for the dev/oidc-provider proejct , used for the ` --with-local-auth` option

Refactors the code to allow for the new version of the oidc-provider library being ESM only (mostly `import` rather than `require`) and modularising the app.

## How should a reviewer test this change?

- clear local auth cookies in your browser
- run `./dev/script/setup --clean --with-local-auth`
- run `./dev/script/start --with-local-auth`
- go to https://media.local.dev-gutools.co.uk/ - you should be redirected to the local-auth ui
- enter the test user id, "[grid-demo-account@guardian.co.uk](mailto:grid-demo-account@guardian.co.uk)" and any password
- click 'authorise' at the next screen
- should be given access and be redirected to the to local grid (https://media.local.dev-gutools.co.uk/).

Note that the `--clean` option on setup removes and recreates your local infastructure, including the local grid DB, so the Grid UI will not have any images to display.



## How can success be measured?
No vulnerabilities from the project

## Who should look at this?


## Tested? Documented?
